src/chat/summarizers/chatHistory.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ChatRequestTurn,
+  ChatResponseMarkdownPart,
+  ChatResponseTurn,
+  LanguageModelTextPart,
+  MarkdownString,
+} from "vscode";
+import { PARTICIPANT_ID } from "../constants";
+import { ToolCallMetadata } from "../tools/types";
+
+export function summarizeChatHistory(
+  history: readonly (ChatRequestTurn | ChatResponseTurn)[],
+): string {
+  let summary = new MarkdownString("These are the previous messages in the conversation:");
+
+  for (const turn of history) {
+    if (turn.participant !== PARTICIPANT_ID) {
+      // skip messages for/from other participants
```

Comment:
This is a nice catch, but it's an icky dev experience to have to do this ourselves